x-pack/plugins/integrations_manager/public/screens/detail.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState, useEffect } from 'react';
+import { EuiPanel, EuiSpacer, EuiTitle } from '@elastic/eui';
+import { getIntegrationInfoByKey } from '../data';
+import { IntegrationInfo } from '../../common/types';
+
+interface MatchPackage {
+  match: {
+    params: {
+      pkgkey: string;
+    };
+  };
+}
+
+export function Detail({ match }: MatchPackage) {
+  const { pkgkey } = match.params;
+  const [info, setInfo] = useState(({} = {} as IntegrationInfo));
```

Comment:
We can just directly supply the type argument to `useState`. This also makes the "not loaded" state explicit by setting it to `null`:

```suggestion
  const [info, setInfo] = useState<IntegrationInfo | null>(null);
```

That requires some extra handling below but should be much safer.